### PR TITLE
fix: update-api-to-match-wdio

### PR DESCRIPTION
**Pull request by FelixZilber**

### CODE REVIEW

File: src/commons/BrowserUtils.ts [hunk 1]
```diff
@@ -53,9 +60,9 @@ export namespace BrowserUtils {
   /**
    * Inject a snippet of JavaScript into the page
    * for execution in the context of the currently selected frame
-   * @param script - js script to execute
+   * @param script - js script to execute in string format
    */
-  export function executeScript(script: string): string {
+  export function execute(script: string): string {
```
Comment (refael-m) [inline src/commons/BrowserUtils.ts:65]: you should add optional arguments parameter

File: src/commons/BrowserUtils.ts [hunk 2]
```diff
@@ -69,18 +76,10 @@ export namespace BrowserUtils {
...
   }
 
-  /**
-   * Get system data tests executed on
-   * Usefully to add in Reporter before each test
-   */
-  export function getSystemData(): string {
-    return String(browser.execute(() => navigator.appVersion));
-  }
-
```
Comment (AlexeyCL) [inline src/commons/BrowserUtils.ts:83]: why did you delete the function?

File: src/commons/BrowserUtils.ts [hunk 3]
```diff
@@ -338,56 +330,40 @@ export namespace BrowserUtils {
    */
   export function switchToFrame(selector: string): void {
```
Comment (AlexeyCL) [inline src/commons/BrowserUtils.ts:331]: selector can be null
```If the frame id is null, the server should switch to the page's default content.```

File: src/commons/BrowserUtils.ts [hunk 4]
```diff
@@ -338,56 +330,40 @@ export namespace BrowserUtils {
...
+    waitForExist(selector);
 
     Reporter.debug(`Switching to an Iframe by selector '${selector}'`);
     tryBlock(() => browser.switchToFrame($(selector)), 'Failed to switch frame');
   }
 
   /**
-   * Switch to other tab by id
+   * The Switch To Window command is used to select the current top-level browsing context for the current session,
+   * i.e. the one that will be used for processing commands.
```
Comment (AlexeyCL) [inline src/commons/BrowserUtils.ts:341]: pls add to jdoc ```a string representing a window handle, should be one of the strings that was returned in a call to getWindowHandles```

File: src/commons/BrowserUtils.ts [hunk 5]
```diff
@@ -709,47 +643,49 @@ export namespace BrowserUtils {
...
    *
    * If no domain provided, will set cookie for current domain
    * Otherwise will first navigate to required domain(should be valid url),
    *  set the cookie and navigate back to page it started from
    * @param cookie cookie to set
    * @param domain domain to set cookie for
    */
-  export function setCookie(cookie: Cookie, domain: string): void {
-    Reporter.debug(`Setting cookie: '${JSON.stringify(cookie)}'`);
+  export function setCookies(cookie: Cookie | Array<Cookie>, domain: string): void {
```
Comment (AlexeyCL) [inline src/commons/BrowserUtils.ts:656]: pls change according ```https://webdriver.io/docs/api/browser/setCookies/```

File: src/commons/BrowserUtils.ts [hunk 6]
```diff
@@ -805,95 +741,78 @@ export namespace BrowserUtils {
...
-      }
-    }
+    tryBlock(() => browser.setWindowSize(width, height), 'Chrome: Failed to resize window');
   }
 
+  /**
+   * Returns browser window size.
+   * output example `{ width: 1280, height: 767 }`
+   */
   export function getWindowSize(): object {
```
Comment (AlexeyCL) [inline src/commons/BrowserUtils.ts:768]: maybe better to return the interface?

File: src/commons/BrowserUtils.ts [hunk 7]
```diff
@@ -805,95 +741,78 @@ export namespace BrowserUtils {
...
-   * @param selector  - element with requested position offset
+   * @param selector element with requested position offset
    */
-  export function getElementLocation(selector: string): Location {
-    Reporter.debug(`Get Element location '${selector}'`);
+  export function getLocation(selector: string): Location {
+    Reporter.debug(`Get element's location '${selector}'`);
 
     return $(selector).getLocation();
   }
```
Comment (AlexeyCL) [inline src/commons/BrowserUtils.ts:818]: accroding to wdio page you should add prop `https://webdriver.io/docs/api/element/getLocation/`

File: src/commons/BrowserUtils.ts [hunk 8]
```diff
@@ -906,15 +825,17 @@ export namespace BrowserUtils {
...
   /**
-   * Send a sequence of key strokes to the active element
+   * Send a sequence of key strokes to the active element.
+   * You can also use characters like "Left arrow" or "Back space".
+   * WebdriverIO will take care of translating them into unicode characters.
    * it can be single key or an array of keys
    * @param keysToSend key, array of keys or string array (chars) to send
    */
-  export function sendKeys(keysToSend: SpecialKeys | Array<SpecialKeys> | string | Array<string>): void {
+  export function keys(keysToSend: SpecialKeys | Array<SpecialKeys> | string | Array<string>): void {
```
Comment (AlexeyCL) [inline src/commons/BrowserUtils.ts:838]: Send a sequence of key strokes to the active element. You can also use characters like "Left arrow" or "Back space". WebdriverIO will take care of translating them into unicode characters. You’ll find all supported characters here. To do that, the value has to correspond to a key from the table.

**Comment by AlexeyCL** [comment]
@FelixZilber well done!!

File: src/commons/BrowserUtils.ts (hunk 1 above)
Comment (FelixZilber) [reply src/commons/BrowserUtils.ts:65]: since we are passing the full script everywhere, and for the sake of simplicity, I left it without arguments

File: src/commons/BrowserUtils.ts (hunk 2 above)
Comment (FelixZilber) [reply src/commons/BrowserUtils.ts:83]: I didn't find usage of it anywere

File: src/commons/BrowserUtils.ts (hunk 3 above)
Comment (FelixZilber) [reply src/commons/BrowserUtils.ts:331]: updated

File: src/commons/BrowserUtils.ts (hunk 4 above)
Comment (FelixZilber) [reply src/commons/BrowserUtils.ts:341]: done

File: src/commons/BrowserUtils.ts (hunk 5 above)
Comment (FelixZilber) [reply src/commons/BrowserUtils.ts:656]: domain handle removed

File: src/commons/BrowserUtils.ts (hunk 6 above)
Comment (FelixZilber) [reply src/commons/BrowserUtils.ts:768]: the BrowserSize is not exported. We can create a PR in  wdio repo with fix

File: src/commons/BrowserUtils.ts (hunk 7 above)
Comment (FelixZilber) [reply src/commons/BrowserUtils.ts:818]: there is a mismatch between docs and actual function signature. anyway don't see a reason to get only part of the data

File: src/commons/BrowserUtils.ts (hunk 8 above)
Comment (FelixZilber) [reply src/commons/BrowserUtils.ts:838]: what's your point here?

Comment (AlexeyCL) [reply src/commons/BrowserUtils.ts:827]: that you can remove keysToSend: SpecialKeys | Array<SpecialKeys>

File: src/commons/BrowserUtils.ts (hunk 1 above)
Comment (refael-m) [reply src/commons/BrowserUtils.ts:65]: ok

**Review by refael-m — APPROVED** [review]

**Review by AlexeyCL — APPROVED** [review]
